cmd/network/verification.go

```diff
@@ -0,0 +1,101 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	ocmlog "github.com/openshift-online/ocm-sdk-go/logging"
+	"github.com/openshift/osd-network-verifier/pkg/cloudclient"
+	"github.com/spf13/cobra"
+)
+
+var (
+	defaultTags            = map[string]string{"osd-network-verifier": "owned", "red-hat-managed": "true", "Name": "osd-network-verifier"}
+	regionEnvVarStr string = "AWS_REGION"
+	regionDefault   string = "us-east-2"
+)
+
+type egressConfig struct {
+	vpcSubnetID  string
+	cloudImageID string
+	instanceType string
+	cloudTags    map[string]string
+	debug        bool
+	region       string
+	timeout      time.Duration
+	kmsKeyID     string
+}
+
+func getDefaultRegion() string {
+	val, present := os.LookupEnv(regionEnvVarStr)
+	if present {
+		return val
+	}
+	return regionDefault
+
+}
+func NewCmdValidateEgress() *cobra.Command {
+	config := egressConfig{}
+
+	validateEgressCmd := &cobra.Command{
+		Use:   "verify-egress",
+		Short: "Verify essential openshift domains are reachable from given subnet ID.",
+		Long:  `Verify essential openshift domains are reachable from given subnet ID.`,
+		Example: `For AWS, ensure your credential environment vars 
+AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY (also AWS_SESSION_TOKEN for STS credentials) 
+are set correctly before execution.
+
+# Verify that essential openshift domains are reachable from a given SUBNET_ID
+osdctl network verify-egress --subnet-id $(SUBNET_ID) --image-id $(IMAGE_ID)`,
+		Run: func(cmd *cobra.Command, args []string) {
+			// ctx
+			ctx := context.TODO()
+
+			// Create logger
+			builder := ocmlog.NewStdLoggerBuilder()
+			builder.Debug(config.debug)
+			logger, err := builder.Build()
+			if err != nil {
+				fmt.Printf("Unable to build logger: %s\n", err.Error())
+				os.Exit(1)
+			}
+
+			logger.Warn(ctx, "Using region: %s", config.region)
+			creds := credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
```

Comment:
What if a user provides a profile? osdctl has `-p` for other AWS commands

Reply:
It's a good idea but should we implement it in ONV (the same cmd exists in ONV) and relay the changes here?

Reply:
added aws profile flag. Depends on https://github.com/openshift/osd-network-verifier/pull/107 . Reconciling with Linh's open PR https://github.com/openshift/osd-network-verifier/pull/106/